### DISCUSSION TITLE
播放进度, 前进后退不准确

### DISCUSSION THREAD

**Discussion by letterk** [hooke007/mpv_PlayKit#13] [Q&A 问答] [answered]
以左右键默认设置5秒为例
快进5秒, 实际可能是6秒, 可能是十几秒, 最夸张的是在播放瑞克和莫蒂, 快进5秒, 实际上快进了50秒. 导致根本无法精确控制时间, 这是咋回事
具体表现就是快进5秒, 播放进度显示5秒, 然后在100ms的时间内一跳, 后面莫名其妙追加了许多秒. 

用过了纯净模式, 依然如此

版本mpv-lazy-20210502, mpv-lazy-20210224

**Comment by hooke007** [comment]
我没有在这方面修改过。
确认是mpv自身的bug的话向官方汇报

======================
本身的默认设置就是定位到最近的关键帧，不过你的描述似乎并不是关键帧的问题。

**Comment by letterk** [comment]
是我草率了, 我发现mpv原版就定位不准, 然后我用potplayer也是一样的问题. 难道是压制的问题?

**Comment by hooke007** [comment]
如果别的视频都正常...那自然是视频本身的问题

**Comment by hooke007** [comment]
修正一下前面一个说法。。自己记混了
对于想要对正常视频也强制精准定位，参考 [#options-hr-seek](https://mpv.io/manual/master/#options-hr-seek)

**Comment by letterk** [comment]
我这里是所有视频都不正常, 在上千个视频中抽查了几十个, 最准确的是5秒变6秒, 最多的是5秒变50秒, 一般的电影基本都是5秒变十几秒, 最扯淡的是一些视频, 前进和后退秒数不一样, 还有一个视频, 后退在2秒和4秒之间交替

**Comment by letterk** [comment]
-hr-seek=yes 参数只对部分视频有效, 而且这样设置后, 跳转速度变慢了, 和直接用鼠标点进度条速度一样, 以前是瞬间跳转, 现在可能要卡几秒

**Comment by hooke007** [comment]
>而且这样设置后, 跳转速度变慢了, 和直接用鼠标点进度条速度一样, 以前是瞬间跳转, 现在可能要卡几秒

这是正常情况符合预期。

>所有视频都不正常

这种情况建议
`mpv --no-config --log-file=mpv.log`
向官方提交issue

**Comment by letterk** [comment]
谢谢你的帮助, 目前至少大部分视频都正常了, 少部分不正常的也不去管他了
因为这个定位问题, 不仅仅是mpv有这个问题, 比如那个5秒变50秒的, 在potplayer上复现问题一模一样. 估计是某个共用的模块出了bug. 不懂英文也就没办法向官方提交issue了

**Comment by kaiqinhu** [comment, marked as answer]
> 我这里是所有视频都不正常, 在上千个视频中抽查了几十个, 最准确的是5秒变6秒, 最多的是5秒变50秒, 一般的电影基本都是5秒变十几秒, 最扯淡的是一些视频, 前进和后退秒数不一样, 还有一个视频, 后退在2秒和4秒之间交替

因为关键帧（I帧）的原因，H264/H265等编码中有I帧、P帧、B帧的区别，I帧是帧内编码，P帧和B帧要参考前/后帧进行编码；当你使用跳转时，播放器首先是定位至最近的关键帧，所以不是精确跳转，一般的视频关键帧的比例都低于1/10。具体可以参考H.264/H.265编码标准相关科普。

**Comment by MC-dusk** [comment]
诶，从pot和VLC转mpv.net_CM，也是没搞明白怎么跳2s不是精确的，原来GUI设置里面没有模板，得自己新建`seek 2 exact`这种项，之前没接触过很容易吊死在`-hr-seek=yes`上面，因为设置里面搜索`精确跳转`只有这项。。
